src/DataProtection/DataProtection/src/Repositories/DefaultKeyStorageDirectories.cs

```diff
@@ -91,8 +89,8 @@ private DefaultKeyStorageDirectories()
         // the %HOME% variable to build up our base key storage path.
         if (!String.IsNullOrEmpty(Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID")))
         {
-            var homeEnvVar = Environment.GetEnvironmentVariable("HOME");
-            if (!String.IsNullOrEmpty(homeEnvVar))
+            var homeEnvVar = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
```

Comment:
I think this is another case where %HOME% may not be the same as this special folder (this is specific to the Azure Websites case).